src/Tasks/Microsoft.NET.Build.Tasks/FrameworkPackages/FrameworkPackages.cs

```diff
@@ -0,0 +1,182 @@
+            if (FrameworkPackagesByFramework.TryGetValue(framework, out var frameworkPackagesForVersion) &&
+                frameworkPackagesForVersion.TryGetValue(frameworkKey, out var frameworkPackage))
+            {
+                frameworkPackages.Add(frameworkPackage);
+            }
+            else
+            {
+                // if we didn't predefine the package overrides, load them from the targeting pack
+                // we might just leave this out since in future frameworks we'll have this functionality built into NuGet.Frameworks
+                var frameworkPackagesFromPack = LoadFrameworkPackagesFromPack(framework, frameworkReference, targetingPackRoot) ?? new FrameworkPackages(framework, frameworkReference);
+
+                Register(frameworkPackagesFromPack);
+
+                frameworkPackages.Add(frameworkPackagesFromPack);
+            }
+        }
+
+        return frameworkPackages.ToArray();
+    }
+
+    private static FrameworkPackages LoadFrameworkPackagesFromPack(NuGetFramework framework, string frameworkName, string targetingPackRoot)
+    {
+        if (framework is null || framework.Framework != FrameworkConstants.FrameworkIdentifiers.NetCoreApp)
+        {
+            return null;
+        }
+
+        var reducer = new FrameworkReducer();
+        var frameworkKey = GetFrameworkKey(frameworkName);
+        var candidateFrameworks = FrameworkPackagesByFramework.Where(pair => pair.Value.ContainsKey(frameworkKey)).Select(pair => pair.Key);
+        var nearestFramework = reducer.GetNearest(framework, candidateFrameworks);
+
+        var frameworkPackages = nearestFramework is null ?
+            new FrameworkPackages(framework, frameworkName) :
+            new FrameworkPackages(framework, frameworkName, FrameworkPackagesByFramework[nearestFramework][frameworkKey]);
+
+        if (!string.IsNullOrEmpty(targetingPackRoot))
+        {
+            var packsFolder = Path.Combine(targetingPackRoot, frameworkName + ".Ref");
+            if (Directory.Exists(packsFolder))
+            {
+                var packVersionPattern = $"{framework.Version.Major}.{framework.Version.Minor}.*";
+                var packDirectories = Directory.GetDirectories(packsFolder, packVersionPattern);
+                var packageOverridesFile = packDirectories
+                                                .Select(d => (Overrides: Path.Combine(d, "data", "PackageOverrides.txt"), Version: ParseVersion(Path.GetFileName(d))))
+                                                .Where(d => File.Exists(d.Overrides))
+                                                .OrderByDescending(d => d.Version)
+                                                .FirstOrDefault().Overrides;
+
+                if (packageOverridesFile is not null)
+                {
+                    // Adapted from https://github.com/dotnet/sdk/blob/c3a8f72c3a5491c693ff8e49e7406136a12c3040/src/Tasks/Common/ConflictResolution/PackageOverride.cs#L52-L68
```

Comment:
Consider moving this logic into a common file that can be used by both tasks.

Reply:
Thanks for pointing that out.  I think I'm going to leave it for now though, as that may make it simpler to take updates to the code from the component-detection repo if we want to do so in the future.  And hopefully before too long we'll switch how this information is delivered and read entirely.